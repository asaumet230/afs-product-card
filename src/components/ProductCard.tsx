import React from 'react';
import { CSSProperties } from 'react';

// Context:
import { ProductContext } from '../context/productContext';

// Hooks:
import { useProduct } from '../hooks/useProduct';

// CSS:
import styles from '../styles/styles.module.css';

// Interfaces:
import { InitialValues, OnChangeArgs, Product, ProductCartHandlers } from '../interfaces/interfaces';



export interface Props {
    product: Product;
    // children?: ReactElement | ReactElement[];
    children: (args: ProductCartHandlers) => JSX.Element
    className?: string;
    style?: CSSProperties;
    onChange?: (args: OnChangeArgs) => void;
    value?: number;
    initialValues?: InitialValues;
}



export const ProductCard = ({ product, children, className, style, onChange, value, initialValues }: Props) => {

    const { Provider } = ProductContext;
    const { counter, increaseBy, reset, maxCount, isMaxCountReached } = useProduct({
        onChange,
        product,
        value,
        initialValues
    });


    return (

        <Provider value={{
            counter,
            product,
            maxCount,
            increaseBy,
        }}>
            <div
                style={style}
                className={`${styles.productCard} ${className}`}>
                {children({
                    product,
                    increaseBy,
                    reset,
                    maxCount,
                    count: counter,
                    isMaxCountReached,
                })}
            </div>
        </Provider>


    )
}

