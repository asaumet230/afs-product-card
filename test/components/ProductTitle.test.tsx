import React from 'react';
import { create } from 'react-test-renderer';

// Component:
import ProductCard, { ProductTitle } from '../../src/components';

// Products;
import { product1 } from '../data/products';


describe('Pruebas en el Componente <ProductTitle />', () => {

    test('Debe de mostrarse el titulo correctamente', () => {

        const wrapper = create(
            <ProductTitle title='Custom product' />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el titulo de producto por defecto', () => {

        const wrapper = create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );


        expect(wrapper).toMatchSnapshot();


    });


});
