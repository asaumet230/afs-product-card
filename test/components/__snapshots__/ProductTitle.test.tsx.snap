// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <ProductTitle /> Debe de mostrar el titulo de producto por defecto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription "
  >
    Coffe-sin-imagen
  </span>
</div>
`;

exports[`Pruebas en el Componente <ProductTitle /> Debe de mostrarse el titulo correctamente 1`] = `
<span
  className="productDescription "
>
  Custom product
</span>
`;
